src/aosm/azext_aosm/cli_handlers/onboarding_sns_handler.py

```diff
@@ -72,9 +72,8 @@ def build_all_parameters_json(self) -> JSONDefinitionElementBuilder:
         for property_name, value in vars(self.config).items():
             params_content[property_name] = value
         base_file = JSONDefinitionElementBuilder(
-            Path(SNS_OUTPUT_FOLDER_FILENAME), json.dumps(params_content, indent=4)
+            Path(SNS_OUTPUT_FOLDER_FILENAME), json.dumps(params_content, indent=4, default=NSDVReferenceConfig.to_dict)

```

Comment:
I like this change, we just need to test that it still publishes successfully

Reply:
I'm not sure that we could apply this change throughout the other handlers, as we add new things that arent in this config at the nfd stage, like arfitact manifest name, but i like this change here

Reply:
I actually changed it to selective params as the NSDV cannot be changed between build and deploy. only parameters like operator_resource_group_name, site_name can be changed